front/lib/actions/mcp_internal_actions/servers/slack.ts

```diff
@@ -99,8 +100,60 @@ function isSlackTokenRevoked(error: unknown): boolean {
   );
 }
 
+// Unknown is expected when we don't have a Slack connection yet
+type SlackAIStatus = "enabled" | "disabled" | "unknown";
+
+const SLACK_AI_STATUS_CACHE_TTL_MS = 5 * 60 * 1000; // 5 minutes
+
+// Cache the result as this involved a call to the Slack API
+const getCachedSlackAIEnablementStatus = cacheWithRedis<
+  SlackAIStatus,
+  [auth: Authenticator, mcpServerId: string]
+>(
+  async (auth: Authenticator, mcpServerId: string): Promise<SlackAIStatus> => {
+    if (!mcpServerId) {
+      return "unknown";
+    }
+
+    const connection = await getConnectionForMCPServer(auth, {
```

Comment:
@Fraggle I'm seeing a couple puzzling things:
1. Once connected at workspace level, I get two calls: one with the valid serverId, and one with a serverId that exists nowhere in mcp_server_connections. I don't know where it's coming from, but it causes some errors when calling getConnectionForMCPServer.
2. Also (and this may be mostly the same thing), after I disconnect, I continue to get calls for the serverId that no longer exists. That makes my caching result in incorrect behavior.

So I suspect there is caching elsewhere that is causing this, and that is not getting invalidated.

Reply:
> hard error or log on this one ?

Just log.

I think it's actually coming from the fact that we don't clean up other tables when an mcp_server_connections is removed. Specifically, we still have orphan entries in mcp_server_views, and I see the unknown id above coming from there.

We also end up with orphans in remote_mcp_server_tool_metadata.